TfIdf_BOW_creator.py

```diff
@@ -0,0 +1,186 @@
+import enchant
+dict = enchant.Dict("en_US")
+from nltk.stem import PorterStemmer
+ps = PorterStemmer()
+
+# this part generate tf-idf BOW
+
+def sentence_processor(sentence):
+    processed_sentence  = ""
+    list_string = sentence.split(' ')
+    for word in list_string:
+        word = word.lower()
+        word = re.sub(r'[^a-z\s]','',word)
+        word = ps.stem(word)
+        if len(word)>2 and dict.check(word):
+            processed_word = word
+        else:
+            processed_word = ""
+
+        processed_sentence+=" "+processed_word
+    return processed_sentence
+
+# Create vocab
+def create_vocab(filename):
+    # open the file in read mode
+    read_file = open(filename, 'r')
+    
+    # creating dictreader object
+    file = csv.DictReader(read_file)
+    
+    # creating empty lists
+    current_sentence = list()
+    is_oligo = list()
+    
+    # iterating over each row and append
+    # values to empty list
+    for col in file:
+        current_sentence.append(col['Current Sentence'])
+        is_oligo.append(col['Is this an oligo?'])
+
+    non_oligo_sentences = list()
+    oligo_sentences = list()
+    non_oligo_vocab = list()
+    oligo_vocab = list()
+
+    for i in range(len(current_sentence)):
+        if is_oligo[i]=="Oligo" and current_sentence[i] not in oligo_sentences: #appends unique sentences only
+            processed_sentence = sentence_processor(current_sentence[i])
+            # oligo_sentences.append(current_sentence[i])
+            oligo_sentences.append(processed_sentence)
+
+            for word in processed_sentence.split():
+                if word not in oligo_vocab:
+                    processed_word = word_processor(word)
+                    if not processed_word: #if not oligo then append that in vocab coz we want that are linked
+                        # with oligo in a sentence, but are themselves not an oligo
+                        oligo_vocab.append(word)
+                        # processed_bow_word = bow_processor(word)
+                        # if not processed_bow_word:
+                        #     oligo_vocab.append(processed_bow_word) 
+
+
+        elif is_oligo[i]!="Oligo" and current_sentence[i] not in non_oligo_sentences:
+            processed_sentence = sentence_processor(current_sentence[i])
+            # oligo_sentences.append(current_sentence[i])
+            non_oligo_sentences.append(processed_sentence)
+
+            for word in processed_sentence.split():
+                if word not in non_oligo_vocab:
+                    processed_word = word_processor(word)
+                    if not processed_word: #if not oligo looking sequence then append that in vocab coz 
+                        # we want that are linked with oligo in a sentence, but are themselves not an oligo
+                        non_oligo_vocab.append(word)
+
+    return non_oligo_sentences, oligo_sentences, non_oligo_vocab, oligo_vocab
+
+
+# Creating an index for each word in our vocab.
```

Comment:
could this be a one-liner with enumerate?
https://book.pythontips.com/en/latest/enumerate.html

Reply:
Enumerate is something different. It's used to get the index and corresponding list element. We are not creating any list here. We want all these values to be returned when it's parent function is called.